mcs/class/System.Windows.Forms/System.Windows.Forms/InputLanguage.cs

```diff
@@ -98,6 +100,31 @@ internal InputLanguage()
 							CultureInfo cultureInfo = CultureInfo.GetCultureInfo (languageId);
 							inputLanguageList.Add (new InputLanguage (IntPtr.Zero, cultureInfo, cultureInfo.Name));
 						}
+					} else if (Environment.OSVersion.Platform == PlatformID.Unix) {
+						// Call the command 'locale -a' to retrieve a list of all available locales.
```

Comment:
do we really need to shell out to `locale` to get this info?

Reply:
I know that it is not the best solution. I have not found any other way to do that. If you know one, I'm interested.

Reply:
Running `locale -a` for me in truss/strace indicates on GNU/Linux it reads from `/usr/share/locale/locale.alias` and `/usr/lib/locale`; don't count on those values being the same, however. On AIX, it reads from `/usr/share/locale/locale.alias`. I'm sure macOS and the BSDs deviate too.

Reply:
Hmm yeah there doesn't seem to be a good way to get them. The only concern I have is that Application.cs has a static field which reads `InputLanguage.CurrentInputLanguage` which in turn ends up reading `InstalledInputLanguages` through `InputLanguage.FromCulture()`.

This means that this method will essentially be called at startup of any WinForms app, i.e. if anything goes wrong here the app will probably break.

I guess I'd be fine if we changed that and only set that field when `Application.CurrentInputLanguage` is accessed for the first time so only apps using that property run this code.

Reply:
For posterity (because I was a little curious about why `locale -a` is a reasonable option):

Here's how `locale -a` is implemented with glibc's `locale` utlity:
https://sourceware.org/git/?p=glibc.git;a=blob;f=locale/programs/locale.c;h=e07108d3451b293df364f95e8500ff68aeb7b797;hb=HEAD#l410

Here's how `locale -a` is implemented on FreeBSD:
https://github.com/freebsd/freebsd/blob/5cac6a7e8075d6b435bf12bf4eea538d63ec8cbd/usr.bin/locale/locale.c#L517

And POSIX doesn't standardize locale enumeration:
https://pubs.opengroup.org/onlinepubs/9699919799/utilities/locale.html
https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap07.html